wallet/wallet.go

```diff
@@ -91,11 +94,20 @@ func NewMultiWalletFromMnemonic(m string, n int) (*Wallet, error) {
 	if n < 0 || n > common.MaxAccountsPerWallet {
 		return nil, fmt.Errorf("invalid number of accounts")
 	}
+
+	// bip39 lib doesn't properly validate whitespace so we have to do that manually.
+	if expected := strings.Join(strings.Fields(m), " "); m != expected {
+		return nil, errWhitespace
```

Comment:
why not simply normalize the input instead? returning the error would be confusing for user
they then have to visually find where the extra white space is?

Reply:
IMHO this is less error-prone. if there's extra whitespace, then the user did something wrong, likely a copy and paste error. i think proceeding to generate a key in spite of this is a bad idea. the user should be aware of the error and should correct it. the BIP39 spec, such as it is, does not allow extra whitespace.

Reply:
but then if there are sure of it (extra whitespace in btwn phrases), they will never be able to use this phrase.

should print the exact diff to user and prompt them to confirm then.

Reply:
in the interest of time, instead of creating more code, maybe just ask user to use the phrase smcli created

Reply:
@countvonzero I think this code should be as simple as possible. I don't think it's the responsibility of the wallet app to correct the user's mistake. At most we could link them to a doc that explains the issue and how to correct it, something along those lines.

Reply:
but what if its not a mistake. and its a phrase they have been using in other wallets?

Reply:
then the phrase is wrong and they should use one that's valid. and they should be made aware that it's wrong.

Reply:
in any case. otherwise code lg

Reply:
I have a related but different question. Whitespace is more than just spaces, does bip39 handle things like `\r` `\n` and `\t` correctly?

Reply:
@fasmat all whitespace is treated the same way. this change will complain about \r, \n, and \t as well, see https://pkg.go.dev/strings#Fields. the regression test checks things other than spaces. the only thing that's permitted is a single space between words.

Reply:
Sorry, didn't see the other whitespace characters in the test. Then LGTM